api/scpca_portal/test/test_loader.py

```diff
@@ -0,0 +1,311 @@
+import re
+import shutil
+from functools import partial
+from zipfile import ZipFile
+
+from django.conf import settings
+from django.test import TransactionTestCase
+
+from scpca_portal import common, loader
+from scpca_portal.models import Project
+from scpca_portal.models.computed_file import ComputedFile
+from scpca_portal.models.library import Library
+from scpca_portal.models.sample import Sample
+
+
+class TestGenerateComputedFiles(TransactionTestCase):
+    def setUp(self):
+        self.get_projects_metadata = partial(
+            loader.get_projects_metadata, input_bucket_name=settings.AWS_S3_INPUT_BUCKET_NAME
+        )
+        # When passing a project_id to get_projects_metadata, a list of one item is returned
+        # This lambda creates a shorthand with which to access the single returned project_metadata
+        self.get_project_metadata = lambda project_id: self.get_projects_metadata(
+            filter_on_project_id=project_id
+        )[0]
+
+        self.create_project = partial(
+            loader.create_project,
+            submitter_whitelist={"scpca"},
+            input_bucket_name=settings.AWS_S3_INPUT_BUCKET_NAME,
+            reload_existing=True,
+            update_s3=False,
+        )
+
+        self.generate_computed_files = partial(
+            loader.generate_computed_files,
+            max_workers=10,
+            update_s3=False,
+            clean_up_output_data=False,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        shutil.rmtree(common.OUTPUT_DATA_PATH, ignore_errors=True)
+
+    def test_create_project_SCPCP999990(self):
+        loader.prep_data_dirs()
+
+        project_id = "SCPCP999990"
+        returned_project = self.create_project(self.get_project_metadata(project_id))
+
+        # CHECK FOR PROJECT EXISTENCE
+        project = Project.objects.filter(scpca_id=project_id).first()
+        self.assertEqual(project, returned_project)
+
+        # CHECK PROJECT ATTRIBUTE VALUES
+        self.assertEqual(project.abstract, "TBD")
+        self.assertEqual(project.additional_restrictions, "Research or academic purposes only")
+        self.assertIn(
+            "SCPCP999990/merged/SCPCP999990_merged-summary-report.html", project.data_file_paths
+        )
+        self.assertIn("SCPCP999990/merged/SCPCP999990_merged.rds", project.data_file_paths)
+        self.assertIn("SCPCP999990/merged/SCPCP999990_merged_rna.h5ad", project.data_file_paths)
+        self.assertIn("SCPCP999990/bulk/SCPCP999990_bulk_metadata.tsv", project.data_file_paths)
+        self.assertIn("SCPCP999990/bulk/SCPCP999990_bulk_quant.tsv", project.data_file_paths)
```

Comment:
Let's do the following:

expected_data_file_paths = [<input vals into list>]
actual_data_file_paths = sorted(project.data_file_paths)